go-controller/pkg/node/openflow_manager.go

```diff
@@ -211,9 +211,15 @@ func (c *openflowManager) Run(stopChan <-chan struct{}, doneWg *sync.WaitGroup)
 }
 
 func (c *openflowManager) updateBridgePMTUDFlowCache(key string, ipAddrs []string) {
+	// protect defaultBridge config from being updated by gw.nodeIPManager
+	c.defaultBridge.Lock()
+	defer c.defaultBridge.Unlock()
+
 	dftFlows := pmtudDropFlows(c.defaultBridge, ipAddrs)
 	c.updateFlowCacheEntry(key, dftFlows)
 	if c.externalGatewayBridge != nil {
+		c.externalGatewayBridge.Lock()
+		defer c.externalGatewayBridge.Unlock()
```

Comment:
Shall we put the lock code into `updateFlowCacheEntry` and `updateExBridgeFlowCacheEntry` respectively?

Reply:
why? we need to cover for use done in `pmtudDropFlows`. Following analogous approach to what we have in `updateBridgeFlowCache`

Reply:
i think the locks are in the right place. Those locks need to happen before pmtudDropFlows, which calls bridge.patchedNetConfigs() and operates on the bridge